src/audio/audio_stream.c

```diff
@@ -21,10 +21,15 @@ static int audio_stream_get_buffer(struct sof_sink __sparse_cache *sink, size_t
 	struct audio_stream __sparse_cache *audio_stream =
 			attr_container_of(sink, struct audio_stream __sparse_cache,
 					  sink_api, __sparse_cache);
+	struct comp_buffer __sparse_cache *buffer_c =
+			attr_container_of(audio_stream, struct comp_buffer __sparse_cache,
+					  stream, __sparse_cache);
 
 	if (req_size > audio_stream_get_free_size(sink))
 		return -ENODATA;
 
+	buffer_stream_invalidate(buffer_c, req_size);
```

Comment:
Will this impact MCPS of current LL code today ? or will this PR move the invalidate from one place to another ?

Reply:
Just move from module to base code

Reply:
ok, can we remove from any modules in same patch to avoid any increase in MCPS e.g. for double invalidates. I'm guessing we have some module still doing this ?

Reply:
I double checked, seems only one place use this:
https://github.com/thesofproject/sof/blob/main/src/audio/sink_api_helper.c#L30
no repeated invalidate operation.

it will have minor mcps impact, since it is in copy path.